monads/src/HW/Eval.hs

```diff
@@ -29,8 +29,48 @@ initialState = MachineState [] M.empty
 -- Execute a single instruction. 
 -- Successful evaluation does not produce any useful result: only the effect of modifying state matters. 
 execInstr :: (Ord v, Show v) => StackInstr v -> MyState (MachineState v) (Either (Error v) ())
-execInstr = undefined 
+execInstr instr = do
+    MachineState stack env <- get
+    case instr of
+        PushNum a -> do
+            put $ MachineState (a : stack) env
+            return $ Right ()
```

Comment:
Whenever you don't touch a part of the state, it's better to use `modify` instead of `get` and `put`: 

```
execInstr (PushNum n) =
  Right <$> modify (\m -> m {getStack = n : getStack m})
```

This way you are less likely to make a mistake.